analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+ref_dir <- file.path("..", "ref_files")
+plots_dir <- file.path("plots", "tcga-vs-pbta-plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+### Special functions 
+
+Function for setting up PBTA data from the database. 
+This database was originally created by the [run consensus bash script](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-pbta.sh)
+
+```{r}
+set_up_pbta <- function(data.name) {
+  # Given the name of dataset in the PBTA database, set up the TCGA data. 
+
+  # Start up connection
+  con <- DBI::dbConnect(
+    RSQLite::SQLite(),
+    file.path(scratch_dir, "snv_db.sqlite")
+  )
+  # Connect to SQL
+  df <- dplyr::tbl(con, data.name) %>%
+    # Only keep the columns we want
+    dplyr::select(cols_to_keep) %>%
+    as.data.frame() %>%
+    # Tack on the metadata columns we want
+    dplyr::inner_join(
+      pbta_metadata %>%
+        dplyr::select(
+          Tumor_Sample_Barcode = Kids_First_Biospecimen_ID,
+          experimental_strategy, short_histology
+        )
+    ) 
+  return(df)
+}
+```
+
+Function for setting up TCGA data from the database. 
+This database was originally created by the [run consensus bash script for tcga](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-tcga.sh)
+
+```{r}
+set_up_tcga <- function(data.name) {
+  # Given the name of dataset in the TCGA database, set up the TCGA data. 
+  
+  # Start up connection
+  con <- DBI::dbConnect(
+    RSQLite::SQLite(), 
+    file.path(scratch_dir, "tcga_snv_db.sqlite")
+    )
+
+  # Connect to SQL
+  df <- dplyr::tbl(con, data.name) %>%
+    # Only keep the columns we want
+    dplyr::select(cols_to_keep) %>%
+    as.data.frame() %>%
+    # Shorten the Tumor_Sample_Barcode so it matches
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12)) %>%
+    # Tack on the metadata columns we want
+    dplyr::inner_join(
+      tcga_metadata %>% 
+        dplyr::select(
+          Tumor_Sample_Barcode, 
+          short_histology,
+          experimental_strategy
+          )
+      )
+
+  return(df)
+}
+```
+
+Function for making a combined CDF plot for TMB. 
+This plotting function was adapted from the [`breaks_cdf_plot` function in the 
+`chromosomal-instability`](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/b1b73fe321a97fa82d85c86d20bd85635aabba25/analyses/chromosomal-instability/util/chr-break-plot.R#L120)
+
+```{r}
+tmb_cdf_plot <- function(tmb_df, plot_title) {
+  # Given a data.frame of TMB data, plot it as a CDF plot and save it as a png. 
+  #
+  # Args:
+  #   tmb_df: a chromosomal breaks density file path where each sample is
+  #                 a row with `samples` and `breaks_count` columns. 
+  #   plot_title: to be used for the ggplot2::ggtitle and what it will be saved 
+  #                as a png as.
+  
+  cdf_plot <- tmb_df %>%
```

Comment:
I wouldn't complain if this function was pulled out to a utility script and sourced as needed. It seems a good general use thing for people to fine easily.

Reply:
This is something I have on my TO-DO list but I think it would be better if it were in the module that has the "real" plot (tmb-compare-tcga) and not with a notebook that will be retired. I will file an issue in my own repo so we have this written down.

Reply:
Tis documented : https://github.com/cansavvy/openpbta-notebook-concept/issues/6